domain/modeldefaults/service/providers.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils/v3/ssh"
+
+	"github.com/juju/juju/environs/config"
+)
+
+// ControllerModelConfigService represents a slimmed down model config service
+// for providing access to the controller models configuration.
+type ControllerModelConfigService interface {
+	ModelConfig(context.Context) (*config.Config, error)
+}
+
+// SystemSSHKeyProvider is responsible for providing the controllers system ssh key that
+// gets added to every model as a default. If no system ssh key is found then a
+// error satisfying [errors.NotFound] will be returned.
+type SystemSSHKeyProvider = func(context.Context) (string, error)
+
+// NewSystemSSHKeyProvider returns a [SystemSSHKeyProvider] that will return the
+// set system ssh key for the controller. The provider finds the key in the
+// controller mo
```

Comment:
typo

Reply:
This file should never have made it in. Done & Done